x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connectors/create_connector/create_connector.tsx

```diff
@@ -67,10 +67,8 @@ export const CreateConnector: React.FC = () => {
 
   useEffect(() => {
     // TODO: separate this to ability and preference
-    if (!selectedConnector?.isNative || !selfManagePreference) {
+    if (selectedConnector && !selectedConnector.isNative && selfManagePreference === 'native') {
       setSelfManagePreference('selfManaged');
-    } else {
-      setSelfManagePreference('native');
```

Comment:
@JoseLuisGJ Why did we remove `setSelfManagePreference('native');` here? What if we have self-managed preference now, and we want to change it back to `native` would the new flow work with your changes?

Reply:
@jedrazb @navarone-feekery @artem-shelkovnikov The context and the goal for that hook is that as you know some connectors are only available as a Self-managed. So that when users have `selfManagePreference == 'native' ` (Choosing this radio button option) and they choose a connector that only works in Self-managed mode, we switch the radio button option to `setSelfManagePreference('selfManaged')`

Sharing here how it works with the current logic vs the expected one.

### Current hook
```
useEffect(() => {
    if (!selectedConnector?.isNative || !selfManagePreference) {
      setSelfManagePreference('selfManaged');
    } else {
      setSelfManagePreference('native');
    }
  }, [selectedConnector]);
```

![CleanShot 2024-10-31 at 07 53 26](https://github.com/user-attachments/assets/bd950554-0acc-43bc-afee-4fd17784c3a3)

Issues:
- When loading the page, by default always switches to Self-managed radio button option even tho we have as a default value. We want to be Elastic managed by default. 
- If we choose a connector in Self-managed mode and choose another different connector which has this mode available, it switches to Elastic managed in the radio button options which causes confusion





### Suggested hook
```
useEffect(() => {
    if (selectedConnector && !selectedConnector.isNative && selfManagePreference === 'native') {
      setSelfManagePreference('selfManaged');
    }
  }, [selectedConnector]);
```

![CleanShot 2024-10-31 at 07 54 44](https://github.com/user-attachments/assets/b574bc5e-8441-4d5d-aab5-65a2c5d75c99)

Comments:
- Only if we have chosen in the radio buttons Self-managed and we choose a connector available as a Self-managed. Only in this case we change the radio button option to Self-managed and we also disable the Elastic managed radio button.
- By default Elastic managed is preselected as expected.

Reply:
ah okok all clear 🙇  your explanation makes sense and now I see the difference between `connector.is_native` and `selfManagePreference` 

Ship it! 🚀